speedtest/server.go

```diff
@@ -212,6 +213,23 @@ func (s *Speedtest) FetchServerListContext(ctx context.Context) (Servers, error)
 		query.Set("lat", strconv.FormatFloat(s.config.Location.Lat, 'f', -1, 64))
 		query.Set("lon", strconv.FormatFloat(s.config.Location.Lon, 'f', -1, 64))
 	}
+
+	if len(s.config.CountryCode) > 0 {
+		var lowerCode = strings.ToLower(s.config.CountryCode)
+		var lat float64 = 0
+		var lon float64 = 0
+		for _, v := range Locations {
+			if v.CC == lowerCode {
+				lat = v.Lat
+				lon = v.Lon
+			}
+		}
+		if lat != 0 && lon != 0 {
+			query.Set("lat", strconv.FormatFloat(lat, 'f', -1, 64))
+			query.Set("lon", strconv.FormatFloat(lon, 'f', -1, 64))
```

Comment:
If the area of a country is too large, the location will be meaningless.
eg: the `s.config.CountryCode = CN` and lat,lon will always directed to Beijing.(It will failed when i want to direct to China South.)